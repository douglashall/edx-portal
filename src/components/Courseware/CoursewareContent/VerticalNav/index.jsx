import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Button } from '@edx/paragon';


const VerticalNav = function VerticalNav(props) {
  return (
    <div>
      <Link to={{
          pathname: props.previousPath,
          state: props.previousNode,
          }}
        >
        <Button label="Previous" />
      </Link>
      
      <Link to={{
          pathname: props.nextPath,
          state: props.nextNode,
          }}
        >
        <Button label="Next" />
      </Link>
    </div>
  );
};

VerticalNav.defaultProps = {
  previousPath: '',
  previousNode: {},
  nextPath: '',
  nextNode: {},
};

VerticalNav.propTypes = {
  previousPath: PropTypes.string,
  previousNode: PropTypes.object, // eslint-disable-line react/forbid-prop-types
  nextPath: PropTypes.string,
  nextNode: PropTypes.object, // eslint-disable-line react/forbid-prop-types
};

export default VerticalNav;
