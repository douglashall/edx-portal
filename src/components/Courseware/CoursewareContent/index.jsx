import React from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';

function CoursewareContent(props) {
  return (
    <div>
      <Helmet>
        <title>Course</title>
      </Helmet>
      <div className="embed-responsive embed-responsive-1by1" key={props.node.id}>
        <iframe src={props.node.displayUrl} title={props.node.displayName} />
      </div>
    </div>
  );
}

CoursewareContent.propTypes = {
  node: PropTypes.shape({
    id: PropTypes.string.isRequired,
    displayUrl: PropTypes.string.isRequired,
    displayName: PropTypes.string.isRequired,
  }).isRequired,
};

export default CoursewareContent;
