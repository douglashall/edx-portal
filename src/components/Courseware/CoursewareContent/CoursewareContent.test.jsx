import React from 'react';
import renderer from 'react-test-renderer';
import CoursewareContent from './index';

describe('CoursewareContent', () => {
  const mockNode = {
    id: 'dummynode',
    displayUrl: 'http://www.example.com',
    displayName: 'Just a dummy node to display',
  };

  it('renders correctly', () => {
    const tree = renderer
      .create(<CoursewareContent node={mockNode} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
