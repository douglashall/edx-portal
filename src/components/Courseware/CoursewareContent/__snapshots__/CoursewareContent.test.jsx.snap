// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoursewareContent renders correctly 1`] = `
<div>
  <div
    className="embed-responsive embed-responsive-1by1"
  >
    <iframe
      src="http://www.example.com"
      title="Just a dummy node to display"
    />
  </div>
</div>
`;
